### PR TITLE
Dealing with dissapear track causing error.

### PR DESCRIPTION
**Pull request by cropse**
In some scenarios, if a track is banned, deleted, or disappears, the playlist will contain an empty track (None). This error occurs as a result but can be easily fixed.
Error code:
```
  File "/config/custom_components/spotcast/__init__.py", line 360, in start_casting
    add_tracks_to_queue(client, searchResults[1:])
  File "/config/custom_components/spotcast/helpers.py", line 353, in add_tracks_to_queue
    filtered = list(filter(lambda x: x["type"] == "track", tracks))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/spotcast/helpers.py", line 353, in <lambda>
    filtered = list(filter(lambda x: x["type"] == "track", tracks))
                                     ~^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

### CODE REVIEW

File: custom_components/spotcast/helpers.py [hunk 1]
```diff
@@ -349,7 +349,7 @@ def search_tracks(
 def add_tracks_to_queue(
     spotify_client: spotipy.Spotify, tracks: list = [], limit: int = 20
 ):
-    filtered = list(filter(lambda x: x["type"] == "track", tracks))
+    filtered = list(filter(lambda x: x and x["type"] == "track", tracks))
```
Comment (fcusson) [inline custom_components/spotcast/helpers.py:352]: I would prefer we test specifically for `dict` type and we should also use the `get` method to ensure cases where the key is potentially missing. Should look like:

```python
filtered = list(filter(lambda x: isinstance(x, dict) and x.get("type") == "track", tracks))
```

**Review by fcusson — CHANGES_REQUESTED** [review]
Small change just to go with the QC I'm trying to implement. Once you push the change, I'll merge for next release

**Comment by fcusson** [comment]
@cropse would you be able to provide an example of a case where this is happening, I want to test this edge case for the current `v5` build

**Comment by cropse** [comment]
Sure, I think it's quite hard to reproduce since it's a really rare condition.
1. Open `Discovery Weekly` play list on Phone (This is probably only playlist which can be hiden so must test with this one)
2. click miner icon to hide the song
3. then log the tracks
4. Found data include `None` in the list.
Trimed e.g. looks like:
```[
    {
        "collaborative": False,
        "description": "Discover &#x27;Mellow Music&#x27; 🎧 Perfect chill vibes for any time.  Updated Weekly...ludwigvanbeethoven586@gmail.com",
        "external_urls": {
            "spotify": "https://open.spotify.com/playlist/7ci4PEose2PtQPCOCCHKlP"
        },
        "href": "https://api.spotify.com/v1/playlists/7ci4PEose2PtQPCOCCHKlP",
        "id": "7ci4PEose2PtQPCOCCHKlP",
        "images": [
            {
                "height": None,
                "url": "https://image-cdn-ak.spotifycdn.com/image/ab67706c0000da84bcbfba1e69d1319f5dbee17f",
                "width": None,
            }
        ],
        "name": "Mellow music♭",
        "owner": {
            "display_name": "Ludwig van Beethoven",
            "external_urls": {
                "spotify": "https://open.spotify.com/user/315jlrz72kxu4bfycvwjlp6cl6k4"
            },
            "href": "https://api.spotify.com/v1/users/315jlrz72kxu4bfycvwjlp6cl6k4",
            "id": "315jlrz72kxu4bfycvwjlp6cl6k4",
            "type": "user",
            "uri": "spotify:user:315jlrz72kxu4bfycvwjlp6cl6k4",
        },
        "primary_color": None,
        "public": True,
        "snapshot_id": "AAAAfZh3P0uioQIrKFbUKri+JDEzGCxG",
        "tracks": {
            "href": "https://api.spotify.com/v1/playlists/7ci4PEose2PtQPCOCCHKlP/tracks",
            "total": 79,
        },
        "type": "playlist",
        "uri": "spotify:playlist:7ci4PEose2PtQPCOCCHKlP",
    },
    None,
    {
        "collaborative": False,
        "description": "bella.7k@outlook.com==The hottest pop tracks that are taking over the charts and social media. Refresh weekly so you will discover something new and exciting.!!",
        "external_urls": {
            "spotify": "https://open.spotify.com/playlist/6AZP6i1KUsgk4GtTlKdDdj"
        },
        "href": "https://api.spotify.com/v1/playlists/6AZP6i1KUsgk4GtTlKdDdj",
        "id": "6AZP6i1KUsgk4GtTlKdDdj",
        "images": [
            {
                "height": None,
                "url": "https://image-cdn-ak.spotifycdn.com/image/ab67706c0000da8448e5228e4b8f1b42e27962b5",
                "width": None,
            }
        ],
        "name": "Viral Vibes",
        "owner": {
            "display_name": "bella.7k",
            "external_urls": {
                "spotify": "https://open.spotify.com/user/314x7fx4s2zrotjoksiq65f7n2de"
            },
            "href": "https://api.spotify.com/v1/users/314x7fx4s2zrotjoksiq65f7n2de",
            "id": "314x7fx4s2zrotjoksiq65f7n2de",
            "type": "user",
            "uri": "spotify:user:314x7fx4s2zrotjoksiq65f7n2de",
        },
        "primary_color": None,
        "public": True,
        "snapshot_id": "AAAAiNu6f2goxoQdeGFThBZDlgXTX5PN",
        "tracks": {
            "href": "https://api.spotify.com/v1/playlists/6AZP6i1KUsgk4GtTlKdDdj/tracks",
            "total": 60,
        },
        "type": "playlist",
        "uri": "spotify:playlist:6AZP6i1KUsgk4GtTlKdDdj",
    },
]

I believe the song got removed should encountered similiar issue but it's hard to test IRL.

File: custom_components/spotcast/helpers.py (hunk 1 above)
Comment (cropse) [reply custom_components/spotcast/helpers.py:352]: This is an edge case for None value, both implementations should be fine for me.

Comment (cropse) [inline custom_components/spotcast/helpers.py:352]: Feel free to commit it.
```suggestion
    filtered = list(filter(lambda x: isinstance(x, dict) and x.get("type") == "track", tracks))
```

**Comment by fcusson** [comment]
for `V5` I have a pager sytem that takes care of retrieving lists from the spotify API. I added a filter at the end to remove any `NoneType` object from the list doing so:

```python
items = [x for x in items if x is not None]
```

That should automatically resolve the issue for any usecase

**Comment by cropse** [comment]
Thanks a lot
